Sources/Workspace/Workspace+State.swift

```diff
@@ -88,6 +88,14 @@ public final class WorkspaceState {
         self.artifacts = storedState.artifacts
         self.prebuilts = storedState.prebuilts
     }
+    
+    public func add(dependency: Workspace.ManagedDependency) {
+        dependencies = dependencies.add(dependency)
+    }
+
+    public func remove(identity: PackageIdentity) {
+        dependencies = dependencies.remove(identity)
+    }
```

Comment:
To keep the setter of `dependencies` `private`, we need to add helper methods for the two operations that mutate `ManagedDependencies`. Alternatively, we could make the setter public, however, I do find this makes callsites of `WorkspaceState` consumers nicer, so I'm leaning to keep the current solution.